funflow/src/Control/FunFlow/Exec/Simple.hs

```diff
@@ -66,12 +70,22 @@ runFlowEx :: forall c eff ex a b. (Coordinator c, Exception ex)
           -> FlowM b
 runFlowEx _ cfg store runWrapped confIdent flow input = do
     hook <- initialise cfg
-    runAsyncA (eval (runFlow' hook) flow) input
+    progressChan :: STM.TChan Progress <- liftIO $ STM.newTChanIO
+    runThread <- lift . async $ runAsyncA (eval (runFlow' hook progressChan) flow) input
```

Comment:
Better use `withAsync` to ensure that e.g. user interrupt takes down the flow runner.

Reply:
Can't actually use `withAsync` here, because it requires that the action monad is the same as the surrounding one. And I can't lift the internal one, because `Stream` doesn't implement `MonadBaseControl`.

Reply:
Right, that's unfortunate. Makes sense though, we're returning a stream that uses the async, so the async needs to persist until the end of the stream.

Nonetheless, the current version is not async exception safe. Looking through the streaming docs they seem to recommend `ResourceT`.